pkg/lib/analytic/chart_service.go

```diff
@@ -14,6 +21,56 @@ func (s *ChartService) GetActiveUserChat(
 	periodical string,
 	rangeFrom time.Time,
 	rangeTo time.Time,
+) (*Chart, error) {
+	countType := ""
+	periodicalType := periodicalutil.Type(periodical)
+	switch periodicalType {
+	case periodicalutil.Weekly:
+		countType = WeeklyActiveUserCountType
+	case periodicalutil.Monthly:
+		countType = MonthlyActiveUserCountType
+	default:
+		return nil, fmt.Errorf("unknown periodical: %s", periodical)
+	}
+
+	dataset, err := s.getDataPointsByCountType(appID, countType, periodicalType, rangeFrom, rangeTo)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Chart{
+		DataSet: dataset,
+	}, nil
+}
+
+func (s *ChartService) getDataPointsByCountType(
+	appID string,
+	countType string,
+	periodical periodicalutil.Type,
+	rangeFrom time.Time,
+	rangeTo time.Time,
 ) ([]*DataPoint, error) {
-	return nil, nil
+
+	counts, err := s.AuditStore.GetAnalyticCountsByType(appID, countType, &rangeFrom, &rangeTo)
+	if err != nil {
+		return nil, err
+	}
+
+	countsMap := map[string]int{}
+	dataPoints := []*DataPoint{}
+	for _, c := range counts {
+		countsMap[c.Date.Format("2006-01-02")] = c.Count
```

Comment:
The constant `"2006-01-02"` is worth defined in `pkg/util/timeutil/layout.go` as

```golang
const LayoutISODate = "2006-01-02"
```